examples/agentInteractionsDemo/util.js

```diff
@@ -0,0 +1,120 @@
+// ----- Utility Functions
+function setErrorState(el, errorMsg) {
+    el.textContent = errorMsg;
+    setHidden(el, false);
+}
+
+function setHidden(el, hidden) {
+    if (hidden) {
+        el.classList.add("hidden");
+    } else {
+        el.classList.remove("hidden");
+    }
+}
+
+function extractParams(paramStr) {
+    let result = {};
+
+    if (paramStr) {
+        let params = paramStr.split('&');
+        params.forEach(function(currParam) {
+            if (currParam) {
+                let paramTokens = currParam.split('=');
+                let paramName = paramTokens[0];
+                let paramValue = paramTokens[1];
+                if (paramName) {
+                    paramName = decodeURIComponent(paramName);
+                    paramValue = paramValue ? decodeURIComponent(paramValue) : null;
+
+                    if (!result.hasOwnProperty(paramName)) {
+                        result[paramName] = paramValue;
+                    } else if (Array.isArray(result[paramName])) {
+                        result[paramName].push(paramValue);
+                    } else {
+                        result[paramName] = [result[paramName], paramValue];
+                    }
+                }
+            }
+        });
+    }
+
+    return result;
+}
+
+/**
+ * Determine the embedding PureCloud environment seeded on the query string or
+ * being returned through the OAuth2 Implicit grant state hash param.
+ *
+ * @returns A string indicating the embedding PC env (e.g. mypurecloud.com, mypurecloud.jp); otherwise, null.
+ */
+function getEmbeddingPCEnv() {
+    let result = null;
+
+    if (window.location.hash && window.location.hash.indexOf('access_token') >= 0) {
+        let oauthParams = extractParams(window.location.hash.substring(1));
+        if (oauthParams && oauthParams.access_token && oauthParams.state) {
+            // OAuth2 spec dictates this encoding
+            // See: https://tools.ietf.org/html/rfc6749#appendix-B
+            let stateSearch = unescape(oauthParams.state);
+            result = extractParams(stateSearch).pcEnvironment;
+        }
+    }
+
+    if (!result && window.location.search) {
+        result = extractParams(window.location.search.substring(1)).pcEnvironment || null;
+    }
+
+    return result;
+}
+
+function processImageData(images) {
```

Comment:
I think these three utilities could also go in app or in separate component files, but this is just an fyi comment.  I don't think we need to make any changes here as this is an example app and we're trying to keep it simple.